### ISSUE TITLE
Format of STRUCT is ambiguous

### ISSUE THREAD

**Issue by apstndb** [cloudspannerecosystem/spanner-cli#104]
Currently, spanner-cli formats of `STRUCT` values as `[...]`.
I think it is ambiguous with format of `ARRAY` values.
(We know nested `ARRAY` and top-level `STRUCT` is forbid so not strictly ambiguous.)

```
$ spanner-cli -e 'SELECT ARRAY(SELECT AS STRUCT 1, 2)'
[[1, 2]]
```

```
$ spanner-cli -e 'SELECT SAMPLE_LOCK_REQUESTS FROM SPANNER_SYS.LOCK_STATS_TOP_HOUR' -t
...
[[tbl._exists, Exclusive], [tbl._exists, Exclusive], [tbl._exists, Exclusive], [tbl._exists, Exclusive]]
...
```

How about does `STRUCT` use `(...)` like native notation?

**Comment by yfuruyama** [comment]
It makes sense, but at the same time it is important to have a consistency with gcloud and Cloud Console.

For example, this is the result of `gcloud spanner databases execute-sql`.
```
$ gcloud spanner databases execute-sql --instance=${INSTANCE} --sql="SELECT ARRAY(SELECT AS STRUCT 1, 2);" ${DB}
(Unspecified)
[['1', '2']]
```

And this is the result by Cloud Console.
![image](https://user-images.githubusercontent.com/1595215/104869222-d4782500-5988-11eb-8e56-980270381a7a.png)

**Comment by apstndb** [comment]
Yeah, it seems there are only bad precedents.
In BigQuery, `STRUCT`(record) values are expanded and `FORMAT("%T", ARRAY(SELECT AS STRUCT 1, 2))` prints `[(1, 2)]` but it isn't yet be supported in Cloud Spanner.

**Comment by yfuruyama** [comment]
Yeah BigQuery's support for STRUCT data type is mature as it supports STRUCT data type for table schema and top-level query result.

How about discussing this again once we hear another user's voice or STRUCT data type gets frequently used in Cloud Spanner?
